service.py

```diff
@@ -0,0 +1,53 @@
+from utils import *
+import time
+
+
+# browse for infinity
+def browse_by_infinity(url: str):
+    start_browse(url)
+    i = 2
+    while True:
+        print("Times: {}".format(i))
+        change_ip()
+        i += 1
+
+
+# browse for times
+def browse_by_times(times: int, url: str):
+    start_browse(url)
+    for i in range(1, times):
+        print("Times: {} / {}".format((i + 1), times))
+        change_ip()
+    tor_quit()
+
+
+# browse for infinity
+def browse_by_close_open(url: str):
```

Comment:
Could the browse_by_infinity method be extended to include an optional argument and some logic to check if the argument as been passed, to then call start_browse() if required? Lines 25 to 31 are very similar to lines 6-12 and the only difference being start_browser() is called instead of change_ip().

Reply:
I already refactor this part to give cmd to control different way.